src/Core/src/Platform/Windows/ImageSourcePartExtensions.cs

```diff
@@ -22,17 +23,21 @@ internal static class ImageSourcePartExtensions
 			if (imageSource == null)
 				return null;
 
+			if (mauiContext == null)
+				return null;
+
 			var events = image as IImageSourcePartEvents;
 
 			events?.LoadingStarted();
 			image.UpdateIsLoading(true);
 
 			try
 			{
-				var service = services.GetRequiredImageSourceService(imageSource);
+				var imageSourceServiceProvider = mauiContext.Services.GetRequiredService<IImageSourceServiceProvider>();
+				var imageSourceService = imageSourceServiceProvider.GetRequiredImageSourceService(imageSource);
 
 				var scale = destinationContext.XamlRoot?.RasterizationScale ?? 1;
```

Comment:
@PureWeen Windows doesn't get scale from the Window. Should it be changed?

Reply:
Possibly related? https://github.com/dotnet/maui/issues/1000

Reply:
Windows has a `GetDisplayDensity()` that gets the display using the HWND, so maybe using that is better?